docs/paths.adoc

```diff
@@ -0,0 +1,148 @@
+
+== Paths
+
+A `path` is a way to access a field or register. It is a compile-time list of
+strings.
+
+=== Construction and manipulation
+
+Paths can be constructed using UDLs which are in the `groov::literals` namespace.
+
+[source,cpp]
+----
+using namespace groov::literals;
+
+auto p1 = "reg"_r;
+auto p2 = "field"_f;
+auto p3 = "group"_g;
+----
+
+NOTE: The `_r` `_f` and `_g` literals are all equivalent; they are intended
```

Comment:
question: if i refer to a register using `_f`, will that result in a compilation error?

Reply:
No - they're all the same.